core/src/main/java/org/apache/iceberg/io/StorageCredential.java

```diff
@@ -29,18 +29,27 @@ public interface StorageCredential extends Serializable {
 
   String prefix();
 
-  SerializableMap<String, String> config();
+  /**
+   * @deprecated since 1.10.0, will be removed in 1.11.0; use {@link #serializableConfig()} instead.
+   */
+  @Deprecated
+  default Map<String, String> config() {
+    return serializableConfig();
+  }
+
+  SerializableMap<String, String> serializableConfig();
```

Comment:
I don't think introducing this helps either, because it also introduces breaking API changes

Reply:
It is not breaking as we deprecate the old interface and adding new interface here. But yeah, revAPI considers new interface addition also as a breaking change. 

I like your approach in #12930